libs/common/include/launchdarkly/data_sources/data_source_status_base.hpp

```diff
@@ -0,0 +1,80 @@
+#include <chrono>
+#include <optional>
+
+#include <launchdarkly/data_sources/data_source_status_error_info.hpp>
+#include <launchdarkly/data_sources/data_source_status_error_kind.hpp>
+
+// Common is included in the namespace to disambiguate from client/server
+// for backward compatibility.
+namespace launchdarkly::common::data_sources {
+
+template <typename TDataSourceState>
+class DataSourceStatusBase {
```

Comment:
Says Base, but it isn't a base now. But it needs a name different than DataSourceStatus, as to not cause an aneurism.